docs/specification.md

```diff
@@ -0,0 +1,555 @@
+  with `UNDER_MIN`.
+- If the swap is successful, the computed converted `Y` tokens will be `transfer`red
+  to the `to_dy` account
+
+```ocaml
+type x_to_y_param = {
+    dx : nat ;
+    deadline : timestamp ;
+    min_dy : nat ;
+    to_dy : address ;
+}
+```
+
+
+### **Y_to_X**
+
+Perform a token [swap](#swaps) from `Y` to `X`.
+
+Note: in order to be able to perform a swap, this contract must be made an
+`operator` or be `approve`d (depending if `Y` is an FA1.2 or FA2 contract) to
+`transfer` the requested amount of tokens.
+
+- The `dy` amount of tokens `Y` is `transfer`ed from the `SENDER` account to
+  this contract's.
+- The [swap fee](#fees) is subtracted from this amount and later awarded to LPs
+  with active positions.
+- When the `Y` token is `ctez`, a [protocol fee](#fees) is subtracted from the
+  tokens being deposited (after the swap fee is applied).
+- If the swap is no longer acceptable because the `deadline` was not met, fails
+  with `TIMED_OUT`.
+- If less than `min_dx` amount of token `X` would be obtained from the swap, fails
+  with `UNDER_MIN`.
+- If the swap is successful, the computed converted `X` tokens will be `transfer`red
+  to the `to_dx` account
+
+```ocaml
+type y_to_x_param = {
+    dy : nat ;
+    deadline : timestamp ;
+    min_dx : nat ;
+    to_dx : address ;
+}
+```
+
+
+### **X_to_X_prime**
+
+Perform a token [swap](#swaps) across two Segmented-CFMM, from this contract's
+`X` to the other's `X` (hereby called `X_prime`).
+
+This entrypoint allows one to make a swap between two tokens that aren't directly
+in a pair, but both of which are paired with the same `Y`.
+
+Note: in order to be able to perform a swap, this contract must be made an
+`operator` or be `approve`d (depending if `X` is an FA1.2 or FA2 contract) to
+`transfer` the requested amount of tokens.
+
+- The `dx` amount of tokens `X` is `transfer`ed from the `SENDER` account to
+  this contract's.
+- The [swap fee](#fees) is subtracted from this amount and later awarded to LPs
+  with active positions.
+- When the `Y` token is `ctez`, a [protocol fee](#fees) is subtracted
+  after converting the `X` tokens to `ctez`.
+- If the swap is no longer acceptable because the `deadline` was not met, fails
+  with `TIMED_OUT`.
+- Instead of completing the single swap, the `x_prime_contract` will be called.
+- If less than `min_dx_prime` amount of token `X_prime` would be obtained from
+  the swap second swap, `x_prime_contract` is expected to fail with `UNDER_MIN`.
+- If the swap is successful, the computed converted `X_prime` tokens are expected
+  to be `transfer`red by the `x_prime_contract` to the `to_dx_prime` account.
+
+```ocaml
+type x_to_x_prime_param = {
+    dx : nat ;
+    x_prime_contract : y_to_x_param contract ;
+    deadline : timestamp ;
+    min_dx_prime : nat ;
+    to_dx_prime : address ;
+}
+```
+
+
+### **Set_position**
+
+Updates or creates a new [position](#positions) in the given range.
+
```

Comment:
It may be worth mentioning, that changing position by 0 is a valid way to obtain the current fees from it.

Reply:
Done